static/markdown/tap-target-maintenance.md

```diff
@@ -3,38 +3,144 @@ title: "Maintaining Taps & Targets"
 
+The Singer community was started by Stitch in 2017 and since then many conventions and best practices have been adopted but are commonly unspoken or not explicitly documented which can be intimidating to newcomers.
+This document is meant to memorialize some of the ways that the Singer community operates to make it easier to join, contribute, and benefit from.
+
+## Best Practices
+
+### Avoid Creating a New Variant When Possible
+
+It's not uncommon in the Singer community (and the open source ecosystem in general) to see people fork a repo in order to add a single feature or bug fix but never get it merged to the upstream repo.
+This approach may solve the short term problem but requires you to take on the burden of maintaining the repo moving forward.
+By contributing back to the upstream you benefit the community and you can free yourself from maintaining your variant moving forward.
+If everyone in our community puts their effort into maintaining the same connectors then we all benefit from less work and better quality.
+
+### Favor Generically Useful vs Custom or Hardcoded Logic
+
+Sometimes it's tempting to build a connector with only your exact needs baked in. For example only adding the streams you need or hard coding organization specific logic like account names, how credentials are sourced, etc. But with a little extra effort you can make it generically useful enough to attract other community members to use and help maintain your code.
+
+- Add as much stream coverage as you can
+- All configs should flow through the config.json input file
+- No hardcoded values and logic, or at least not by default
+
+### Document Settings Well
+
+The most challenging part of using a community maintained connector is configuring it properly.
+Always make sure to document your configurations well.
+
+SDK based connectors have a [--about command option](https://sdk.meltano.com/en/latest/implementation/cli.html#about) that prints out the documented settings and descriptions from `tap.py`.
+Meltano Hub uses these settings and descriptions for documentation, so ensuring they're up to date makes it more useful for everyone.
+
+For non-SDK based connectors you will commonly find an example config.json in the repo.
+The Meltano community has also invested in documenting these on MeltanoHub for your convenience.
+
+### Advertise It On Meltano Hub
+
+Once your connector is functioning and ready to go, make sure you get it listed on Meltano Hub (see [docs](/add-a-tap)) so others can find it.
+More users means more contributors and maintainers for your code base!
+
+### Use the Meltano Singer SDK
+
+Using the SDK allows you to reduce code, increase reliability, comply with the Singer Spec, gain features with an SDK version upgrade, and work well within the Meltano community. The SDK is maintained by Meltano and receives regular updates. Upgrading a tap built on the SDK is much easier than one that isn't.
```

Comment:
I wanted to add something specifically about the benefits of the new testing framework, and I tried to match the tone of the paragraph. Feel free to wordsmith as you see fit.

```suggestion
Using the SDK allows you to reduce code, increase reliability, comply with the Singer Spec, gain features with an SDK version upgrade, and work well within the Meltano community. The SDK is maintained by Meltano and receives regular updates. The SDK also has tons of [valuable pre-written tests](https://sdk.meltano.com/en/latest/testing.html) that you can take advantage of with only a few lines of code, allowing you to maintain and upgrade your tap or target with more confidence and less manual testing effort.
```